### PR TITLE
Adding eval script

### PR DESCRIPTION
**Pull request by calebrob6**
Adds `evaluate.py` for running a model checkpoint file through the validation and test splits of a DataModule and recording the results. This is necessary as `train.py` doesn't save test results, and (even if it did save test results) it doesn't use the best model checkpoint (i.e. the one saved by the `ModelCheckpoint` callback). This also gives us the flexibility to evaluate more than just the metrics recorded at training time.

 Note, this depends on #234.

### CODE REVIEW

File: evaluate.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    RegressionTask,
+    SemanticSegmentationTask,
+)
+from torchgeo.trainers.chesapeake import ChesapeakeCVPRSegmentationTask
+from torchgeo.trainers.landcoverai import LandCoverAISegmentationTask
+from torchgeo.trainers.naipchesapeake import NAIPChesapeakeSegmentationTask
+from torchgeo.trainers.resisc45 import RESISC45ClassificationTask
+from torchgeo.trainers.so2sat import So2SatClassificationTask
+
+TASK_TO_MODULES_MAPPING: Dict[
```
Comment (adamjstewart) [inline evaluate.py:41]: This stuff could just be imported from `train` to reduce code duplication

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Needs integration tests (similar to `tests/test_train.py`)

File: evaluate.py (hunk 1 above)
Comment (calebrob6) [reply evaluate.py:41]: can this all go somewhere in the torchgeo namespace? maybe in __init__.py of trainers?

Comment (adamjstewart) [reply evaluate.py:41]: Sure!

**Comment by calebrob6** [comment]
> Needs integration tests (similar to tests/test_train.py)

Any suggestions for what this looks like given that the script needs pretrained models and downloaded datasets to work?

**Comment by adamjstewart** [comment]
> Any suggestions for what this looks like given that the script needs pretrained models 

I would just instantiate an untrained model, create a model checkpoint, and then use it in the tests.

> and downloaded datasets to work?

just use the same `tests/data` datasets as we use for `train.py`

**Comment by calebrob6** [comment]
> I would just instantiate an untrained model, create a model checkpoint, and then use it in the tests.

I think this is not trivial actually. It seems that saving checkpoints is handled exclusively by the Trainer object in PyTorch Lightning. 

Are you able to look at this sometime later? I don't have the bandwidth to fight with integration tests for awhile.

**Comment by adamjstewart** [comment]
> It seems that saving checkpoints is handled exclusively by the Trainer object in PyTorch Lightning.

@isaaccorley wrote a really useful testing fixture to create fake checkpoints, should be easy to expand if we need to.

> Are you able to look at this sometime later? I don't have the bandwidth to fight with integration tests for awhile.

I probably won't get to this until after #144 is done. Since this script doesn't currently get installed with the library, I'm fine with merging this now and adding tests in the future.

**Review by adamjstewart — DISMISSED** [review]

**Pull request merged**